kubernetes/proposals/split-k8s-listener.md

```diff
@@ -0,0 +1,19 @@
+# This is a design proposal for changes in k8s-listener.
+
+Roadmap outlined by Juergen suggests we split k8s-listener.py into 2 standalone components.
+First component is to run on Kubernetess master node and dispatch orders to second component,
+running on every kube-node.
+
+* Proposed change is to replace `policy_update` call found in `process` function with
+`dispatch_orders` call that will retrieve ip addresses of all hosts from romana topology
+service, then loop over this addresses and send policy object with romana metadata to
+each host via  HTTP POST.
+* Introduce optional commadline parameter that would trigger `ListenAndServe` function in `main`
+instead of kube-api listener. The only HTTP handler for `ListenAndServe` will accept POST requests,
+parse out `policy_definition` from request and call `policy_update`.
+
+# Open questions
+* I haven't got final response about starting this in Golang right away. Feedback was positive 
+during last meeting but last email from Juergen emphasizes using python to save time. Need final decision.
+* Should we relay on romana topology as a source of nodes ip addresses or on kube api? Both have required
```

Comment:
Yes, that's interesting. For now, though, just go with Topology. It's our components trying to communicate with our other components, so it should take inventory from our sources. Consider that in the future we might have adapters that feed environment specific topology information to the Topology service.

At any rate, Topology is going to be good enough for now.